src/commands/push.ts

```diff
@@ -62,7 +54,10 @@ export default async (options: CommandOption): Promise<void> => {
 
       console.log(LOG.PUSHING);
       await pushFiles();
-    });
+    };
+    const watcher = chokidar.watch(rootDir, {persistent: true, awaitWriteFinish: true, ignoreInitial: true});
+    watcher.on('ready', pushFiles); // Push on start
+    watcher.on('all', watchCallback);
```

Comment:
Not sure events like `addDir` and `unlinkDir` are worth processing for they relate to folder events. Inside `watchCallback` I filter and keep only `add`, `change` and `unlink` events (or register `watchCallback` only for those `.on()` events)

Reply:
Done. In practice, it's not much of an issue since default clasp ignore settings ends up filtering out directories. In any case, changed it to just add/change/unlink + added debouncing to handle 'save all' type actions in IDEs so the events get coalesced into a single push.

Reply:
Seeing how you added `debounce`, I guess you use/test the watch feature. There's just too much I did refactor without thoroughly testing it.